src/tiles/hover/hover-pane-controller.ts

```diff
@@ -0,0 +1,463 @@
+
+  /** The timer ID for recognizing a long press event */
+  private longPressTimer?: number;
+
+  /** A record of the last mouse position on the host element, for positioning the hover pane */
+  private lastPointerClientPos = { x: 0, y: 0 };
+
+  constructor(
+    /** The host element to which this controller should attach listeners */
+    private readonly host: ReactiveControllerHost &
+      HoverPaneProviderInterface &
+      HTMLElement,
+    /** Options for adjusting the hover pane behavior (offsets, delays, etc.) */
+    options: HoverPaneControllerOptions = {}
+  ) {
+    this.host.addController(this);
+
+    this.mobileBreakpoint = options.mobileBreakpoint ?? this.mobileBreakpoint;
+    this.offsetX = options.offsetX ?? this.offsetX;
+    this.offsetY = options.offsetY ?? this.offsetY;
+    this.showDelay = options.showDelay ?? this.showDelay;
+    this.hideDelay = options.hideDelay ?? this.hideDelay;
+    this.longPressDelay = options.longPressDelay ?? this.longPressDelay;
+    this.enableLongPress = options.enableLongPress ?? this.enableLongPress;
+  }
+
+  hostConnected(): void {
+    this.attachListeners();
+  }
+
+  hostDisconnected(): void {
+    this.detachListeners();
+  }
+
+  hostUpdated(): void {
+    this.hoverPane = this.host.getHoverPane();
+    this.hoverPaneProps = this.host.getHoverPaneProps();
+  }
+
+  /** @inheritdoc */
+  getTemplate(): HTMLTemplateResult | typeof nothing {
+    return this.shouldRenderHoverPane
+      ? html` ${this.touchBackdropTemplate}
+          <tile-hover-pane
+            .model=${this.hoverPaneProps?.model}
+            .baseNavigationUrl=${this.hoverPaneProps?.baseNavigationUrl}
+            .baseImageUrl=${this.hoverPaneProps?.baseImageUrl}
+            .loggedIn=${this.hoverPaneProps?.loggedIn}
+            .sortParam=${this.hoverPaneProps?.sortParam}
+            .collectionNameCache=${this.hoverPaneProps?.collectionNameCache}
+          ></tile-hover-pane>`
+      : nothing;
+  }
+
+  /**
+   * Produces a template for the invisible touch capture backdrop that
+   * is used to cancel the hover pane on touch devices. We want any
+   * touch interaction on the backdrop to remove the hover pane, and
+   * we don't want to bubble up mouse events that would otherwise
+   * affect the state of the hover pane (e.g., fading it back in).
+   */
+  private get touchBackdropTemplate(): HTMLTemplateResult | typeof nothing {
+    return this.isTouchEnabled && this.enableLongPress
+      ? html`<div
+          id="touch-backdrop"
+          @touchstart=${this.handleBackdropInteraction}
+          @touchmove=${this.handleBackdropInteraction}
+          @touchend=${this.handleBackdropInteraction}
+          @touchcancel=${this.handleBackdropInteraction}
+          @mouseenter=${(e: MouseEvent) => e.stopPropagation()}
+          @mousemove=${(e: MouseEvent) => e.stopPropagation()}
+          @mouseleave=${(e: MouseEvent) => e.stopPropagation()}
+        ></div>`
+      : nothing;
+  }
```

Comment:
Nice work on the touchBackdrop, allowing for cancelling the modal by touching _anywhere_ off the pop-up. 👍🏻